src/Tools/dotnet-monitor/EndpointInfo/ClientEndpointInfoSource.cs

```diff
@@ -68,7 +68,7 @@ public async Task<IEnumerable<IEndpointInfo>> GetEndpointInfoAsync(CancellationT
 
             await Task.WhenAll(endpointInfoTasks);
 
-            return endpointInfoTasks.Where(t => t.Result != null).Select(t => t.Result);
+            return endpointInfoTasks.Where(t => t.Result != null).Select(t => t.Result).Cast<IEndpointInfo>();
         }
     }
```

Comment:
Does cast throw for nulls? If so, then can the original List be non-nullable?

Reply:
Discussed offline, refactoring this code is out of scope of this PR